### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by zmoon**
I put together an initial docs build, addressing #60 and #13.

* new Conda env `xmovie-docs` for docs build placed in `ci/`
  - some of the things can be removed since the nbs don't need to be run
* using sphinx-book-theme like xarray
* some API references for the public stuff, adding to the docstrings
* example gifs/pics currently used in the readme now all generated by the main example nb
  - probably a lot of the readme examples could be removed since they are duplicated in the quickstart nb

To build the docs locally to check it out:
```
conda env create -f ci/environment-docs.yml
conda activate xmovie-docs
cd docs
sphinx-autobuild . _build/html
```

### Remaining:
* [x] RTD configuration and link to docs in readme (maybe as the GH page website as well)
* [x] remove most stuff from readme that is duplicated by the quickstart nb
* ~~more docstrings?~~

### CODE REVIEW

**Comment by jbusecke** [comment]
Hey @zmoon, first of all sorry for the late reply. I was just completely swamped with other projects. Is this PR ready for review or can I help with anything?

**Comment by zmoon** [comment]
Hi @jbusecke yeah I think it is ready for you to check it out. I didn't do this yet:

> probably a lot of the readme examples could be removed since they are duplicated in the quickstart nb

I think I was waiting to see what you thought

Also have to configure ReadTheDocs and add link to readme, etc., but can do that after you review the initial docs.

**Comment by zmoon** [comment]
Also, for the rotating globe examples, I think a dataset that covers the whole globe would be more exciting, maybe you have something in mind that would be easy to use?

**Comment by jbusecke** [comment]
I think it would be nice to get started with a smaller example and then build on it with further PRs?

I just activated RTD builds for PRs, but I am not sure if it will run if I just restart the GH actions (which I just did).

**Comment by jbusecke** [comment]
It seems like it is not triggering. So I guess that is even more reason to merge this now and see how it looks haha.

File: ci/environment-docs.yml
Comment (jbusecke) [inline ci/environment-docs.yml:27]: I am not sure about this part here. I would prefer to add a short contributor guide, with instructions on how to set up the env and then install xmovie with `pip install -e .`, because this here might mess with RTD?

File: docs/conf.py
Comment (jbusecke) [inline docs/conf.py:27]: This should be changed to `xmovie maintainers` since it is not just me anymore.

File: docs/whats-new.rst
Comment (jbusecke) [inline docs/whats-new.rst:9]: You should also give yourself credit here!

File: xmovie/core.py
Comment (jbusecke) [inline xmovie/core.py:498]: Wow this is a really thorough correction all over the docstrings. Thank you so much for all the efforts.

**Review by jbusecke — CHANGES_REQUESTED** [review]
Thanks a ton @zmoon for all these changes. I really appreciate all the work you put into this (I only just now realized how much is fixed here 🤪). 

I made a few suggestions. Once those are implemented I will build it locally and then merge?

**Comment by zmoon** [comment]
> It seems like it is not triggering. So I guess that is even more reason to merge this now and see how it looks haha.

Yeah, I don't think it will until the config file is in the default branch. And then if you want to build for PRs you have to [enable in your settings](https://docs.readthedocs.io/en/stable/pull-requests.html).

File: ci/environment-docs.yml
Comment (zmoon) [reply ci/environment-docs.yml:27]: The issue is that `xmovie` needs to be installed or at least importable since I am loading the version from it. This is a pretty foolproof method. I can confirm it is valid Conda env spec and doesn't mess with RTD.
Xarray [does the same thing](https://github.com/pydata/xarray/blob/f76ea09bebba35a175868a81845c426890ec0330/ci/requirements/doc.yml#L37-L41).
I do think contributor guide with instructions for setting up the envs for normal work and docs work would be a good addition!

**Comment by zmoon** [comment]
> I think it would be nice to get started with a smaller example and then build on it with further PRs?

Ok yeah, future PR for changing or adding example makes sense, just wanted to mention it.

**Comment by jbusecke** [comment]
> Yeah, I don't think it will until the config file is in the default branch. And then if you want to build for PRs you have to enable in your settings.

Definitely did that, but I think I have encountered this before. Ill merge this once the changes are implemented.

File: xmovie/core.py
Comment (zmoon) [reply xmovie/core.py:498]: Welcome!

**Comment by zmoon** [comment]
@jbusecke I think it is ready for you to check out the build locally and let me know what you think

**Comment by jbusecke** [comment]
Just built it locally and it looks great! Ill merge this now and we can see how it looks on RTD

**Comment by zmoon** [comment]
[Looks like](https://readthedocs.org/projects/xmovie/builds/15538923/) it got `Command killed due to excessive memory consumption` during the conda create. Are you able to request a rebuild?

**Comment by jbusecke** [comment]
I also have the option to do this: 
image

But I am not quite sure about the implications. Do you have any experience with this @zmoon ?

**Comment by zmoon** [comment]
> Oh darn. I retriggered the build, but I am pretty sure it will fail again. Are we executing the notebooks on RTD? We might want to switch back to just pushing rendered notebooks...xmovie might be a bit resource hungry for the RTD resources?

By default nbsphinx won't execute the nbs unless they are output-free. So we don't have to worry about that (they are stored in the repo with the outputs). That first failure was during the Conda env creation; it didn't even get to the docs build. And I see this time it has been stuck there for a while too. Given that we are not executing the nbs, maybe after #71 I can create a minimal `requirements.txt` for RTD to use to make this part faster.

I haven't tried the "Remove build environment for *latest*, but it doesn't sound like it would hurt anything.

**Comment by jbusecke** [comment]
I think reducing the dependencies is a good thing, no matter what. We could also try to use mamba to solve this issue. I believe I have done that in another project of mine (Ill dig through some repos later).

**Comment by zmoon** [comment]
Also, I didn't properly link them, but I believe you could close #13 and #60 now

**Pull request merged**
